src/platform/qt/GDBWindow.cpp

```diff
@@ -47,6 +49,28 @@ GDBWindow::GDBWindow(GDBController* controller, QWidget* parent)
 	connect(m_bindAddressEdit, &QLineEdit::textChanged, this, &GDBWindow::bindAddressChanged);
 	settingsGrid->addWidget(m_bindAddressEdit, 1, 1, Qt::AlignLeft);
 
+	QGroupBox* watchpointsSettings = new QGroupBox(tr("Write watchpoints behavior"));
+	mainSegment->addWidget(watchpointsSettings);
+
+	QVBoxLayout* watchpointsSettingsLayout = new QVBoxLayout;
+	watchpointsSettings->setLayout(watchpointsSettingsLayout);
+
+	QButtonGroup* watchpointsButtonGroup = new QButtonGroup(watchpointsSettings);
+	connect(watchpointsButtonGroup, &QButtonGroup::idClicked, controller, &GDBController::setWatchpointsBehavior);
+
+	m_watchpointsStandardRadio = new QRadioButton("Follow GDB's standards", watchpointsSettings);
+	m_watchpointsStandardRadio->setChecked(true);
+	watchpointsButtonGroup->addButton(m_watchpointsStandardRadio, 0);
+	watchpointsSettingsLayout->addWidget(m_watchpointsStandardRadio);
+
+	m_watchpointsOverrideLogicRadio = new QRadioButton("Override GDB's handling logic", watchpointsSettings);
+	watchpointsButtonGroup->addButton(m_watchpointsOverrideLogicRadio, 1);
+	watchpointsSettingsLayout->addWidget(m_watchpointsOverrideLogicRadio);
+
+	m_watchpointsOverrideLogicAnyWriteRadio = new QRadioButton("Override GDB's handling logic and\nbreak on nonmodifying writes", watchpointsSettings);
+	watchpointsButtonGroup->addButton(m_watchpointsOverrideLogicAnyWriteRadio, 2);
```

Comment:
Can you not pass the enum values without casting?

Reply:
I tried and it's not possible. Now I don't exactly know how you would do this with a lambda becasue I am connecting directly to the controller object meaning that if I don't cast on the controller side, I HAVE to cast on the window side, just to be sure, is the later what you would prefer? Cause then I do lambda slot and change the controller function to just be a normal setter.

Reply:
I'd call this resolved now.